### PR TITLE
Topic 96 - WebSocketCloseCode (rest of Lifecycle enumerations)

### LINKED ISSUE #96: Lifecycle Enumerations
Implement https://github.com/Palakis/obs-websocket/blob/master/docs/generated/protocol.md#enumerations

Update tests to check against the enums rather than raw codes and messages (where relevant)

- [x] RequestStatus 
- [x] WebSocketCloseCode 
- [x] EventSubscription

### PR DESCRIPTION
**Pull request by TinaTiel**
Added the [WebSocketCloseCode enum](https://github.com/Palakis/obs-websocket/blob/master/docs/generated/protocol.md#websocketclosecode-enum)
For now I've put it in the same place as the Communicator and Remote, since it doesn't strictly have a home nested in those classes or the listeners.

closes #96

### CODE REVIEW

File: src/main/java/io/obswebsocket/community/client/WebSocketCloseCode.java
Comment (Pjiesco) [inline src/main/java/io/obswebsocket/community/client/WebSocketCloseCode.java:5]: If this is internal only (i'm guessing internal only in the obs-websocket) is it necessary that we add this then?

**Review by Pjiesco — DISMISSED** [review]

File: src/main/java/io/obswebsocket/community/client/WebSocketCloseCode.java
Comment (TinaTiel) [reply src/main/java/io/obswebsocket/community/client/WebSocketCloseCode.java:5]: That is true, however I included it because z is a common enough 'default' that I think it would be wiser to include it / be transparent.

Comment (Pjiesco) [reply src/main/java/io/obswebsocket/community/client/WebSocketCloseCode.java:5]: Sure

Comment (ChristopheCVB) [reply src/main/java/io/obswebsocket/community/client/WebSocketCloseCode.java:5]: Agreed, let's just be as transparent as possible

**Review by Pjiesco — APPROVED** [review]

**Comment by TinaTiel** [comment]
Don't merge yet, please. 
Working on additional changes

**Comment by TinaTiel** [comment]
Okay, I've finished making the other changes -- should probably be reviewed for those before merging. 

- enforced use of the WebSocketCloseCode enum in the listener APIs
- found and cleaned up redundant and/or missing tests in the 'unsecured' suite.

**Comment by TinaTiel** [comment]
Weird, E2E test is stuck queueing...I'm going to stop and re-run it.

**Review by ChristopheCVB — DISMISSED** [review]

**Comment by TinaTiel** [comment]
Holup...one more test

**Comment by ChristopheCVB** [comment]
> Holup...one more test

I'll let you merge ;)

**Comment by TinaTiel** [comment]
That last effort/commit has me feeling a bit salty towards Jetty WebSocketClient. It has several final methods so it cannot be directly mocked. The typical workaround would be to implement the same interface it implements, except that this also cannot be done because it both implements an interface and extends another class. We could facade it out, but it's not a quick work because the methods are proprietary to that client...so we'd need to consider our own facade carefully, and do internal mapping/testing for that (_assuming_ we can at least spy the Jetty client). Just...nope.

**Pull request merged**
